src/api/records.test.tsx

```diff
@@ -0,0 +1,167 @@
+import { Record } from '../app.types';
+import {
+  testRecords,
+  testRecordRows,
+  hooksWrapperWithProviders,
+  getInitialState,
+} from '../setupTests';
+import axios from 'axios';
+import { renderHook, waitFor } from '@testing-library/react';
+import { useRecordCount, useRecordsPaginated } from './records';
+import { PreloadedState } from '@reduxjs/toolkit';
+import { RootState } from '../state/store';
+
+describe('records api functions', () => {
+  let mockData: Record[];
+  let params: URLSearchParams;
+  let state: PreloadedState<RootState>;
+
+  beforeEach(() => {
+    mockData = testRecords;
+    params = new URLSearchParams();
+    state = getInitialState();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('useRecordsPaginated', () => {
+    beforeEach(() => {
+      (axios.get as jest.Mock).mockResolvedValue({
+        data: mockData,
+      });
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('sends axios request to fetch records and returns successful response', async () => {
+      params.append('limit', '25');
+      params.append('skip', '0');
+
+      const { result } = renderHook(() => useRecordsPaginated(), {
+        wrapper: hooksWrapperWithProviders(),
+      });
+
+      await waitFor(() => {
+        expect(result.current.isSuccess).toBeTruthy();
+      });
+
+      expect(axios.get).toHaveBeenCalledWith(
+        '/records',
+        expect.objectContaining({ params })
+      );
+      expect(result.current.data).toEqual(testRecordRows);
+    });
```

Comment:
@louise-davies I know you'll be amazed this still isn't merged but I thought I'd have one more go at things to try and get the test perfect here.

I've got it working and we can now do direct `expect(x).toEqual(y)` comparisons on this line but at a cost. You'll notice in the select statement of `useRecordsPaginated` that when we have an image or waveform channel, instead of setting the channel data to a raw `<img>` element, I now set it to its stringified equivalent. The raw `<img>` element was introducing difficulties that prevented an easy comparison to an expected result during tests. Now I can just compare strings here.

This means we have to now detect the data type of the channel in `dataCell.component.tsx` in order to render it properly. If the data type is an image or waveform, parse the stringified `<img>` with a third-party HTML parser and render it. Otherwise, just display the passed-in `rowData`. Because of the changes to how `rowData` is passed to the `dataCell`, I also had to change where scalar number rounding happened. This code has also been moved to `dataCell`.

Obviously this is a fair bit of upheaval so I'd appreciate again if you could review the new work. If you think this is not worth the effort, I'm happy for us to roll back to what was previously approved. Either way, I'd like this PR finally merged ASAP so when you see this comment, either approve and merge the new work or roll back to previous approval and merge that!

Reply:
Following discussion, it was decided to remove this method as its not worth the confusion and goes against React Table methodology. I have now reverted back to commit https://github.com/ral-facilities/operationsgateway/pull/54/commits/40212ac76ae29bca05b20b935c80823da3c75abd